hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
Minimal example:

```console
$ cat empty-array.sh 
#!/bin/bash
set -o nounset
declare -a test_packages
echo $tests_packages
$ ./empty-array.sh 
./empty-array.sh: line 4: tests_packages: unbound variable
```

If we "declare but don't set", we get unbound variable on first usage.

And also, related, unsetting the 0-th element of an array also unsets the array and this is one of the reasons to always keep some dummy value in the 0-th position. The other reason is that we don't want to produce empty strings when accessing `${array[@]}`, for `go test '' github.com/openshift/origin/pkg/build/api` tries to test the current directory (empty string) and the build/api. Instead, we use `${array[@]:1}` that "just works".